posthog/api/hog_function.py

```diff
@@ -433,3 +451,73 @@ def perform_update(self, serializer):
                 changes=changes, name=serializer.instance.name, type=serializer.instance.type or "destination"
             ),
         )
+
+    @action(methods=["PATCH"], detail=False)
+    def rearrange(self, request: Request, *args, **kwargs) -> Response:
+        """Update the execution order of multiple HogFunctions."""
+        team = self.team
+        orders: dict[str, int] = request.data.get("orders", {})
+
+        if not orders:
+            raise exceptions.ValidationError("No orders provided")
+
+        with transaction.atomic():
+            # Get all functions in a single query and validate them
+            function_ids = list(orders.keys())
+            functions = {
+                str(f.id): f
+                for f in HogFunction.objects.filter(
+                    id__in=function_ids, team=team, type="transformation", deleted=False
+                )
+            }
+
+            # Validate all functions exist
+            missing_ids = set(function_ids) - set(functions.keys())
+            if missing_ids:
+                raise exceptions.ValidationError(f"HogFunction with id {missing_ids.pop()} does not exist")
+
+            # Update orders and create activity logs
+            from django.utils import timezone
+
+            current_time = timezone.now()
+
+            for function_id, function in functions.items():
+                new_order = orders[function_id]
+                old_order = function.execution_order
+
+                if old_order != new_order:
+                    function.execution_order = new_order
+                    function.updated_at = current_time
+
+                    log_activity(
+                        organization_id=self.organization.id,
+                        team_id=self.team_id,
+                        user=request.user,
+                        item_id=str(function.id),
+                        was_impersonated=is_impersonated_session(request),
+                        scope="HogFunction",
+                        activity="updated",
+                        detail=Detail(
+                            name=function.name,
+                            type="transformation",
+                            changes=[
+                                Change(
+                                    type="transformation",
```

Comment:
I think here the type is the Model type in this case "HogFunction"
```suggestion
                                    type="HogFunction",
```